conn/i2c/i2c.go

```diff
@@ -83,39 +85,123 @@ func (d *Dev) Write(b []byte) (int, error) {
 	return len(b), nil
 }
 
-// ReadReg writes the register number to the I²C bus, then reads data.
-//
-// It's a wrapper for Tx()
-func (d *Dev) ReadReg(reg byte, b []byte) error {
+// DevReg is a Dev that exposes memory mapped registers.
+type DevReg struct {
+	Dev
+	// Order specifies the binary encoding of words. It is expected to be either
+	// binary.BigEndian or binary.LittleEndian or a specialized implemented if
+	// necessary. A good example of such need is devices communicating 32bits
+	// little endian values encoded over 16bits big endian words.
+	Order binary.ByteOrder
+}
+
+// ReadRegRaw writes the register number to the I²C bus, then reads data
+// unmarshalled.
+func (d *DevReg) ReadRegRaw(reg byte, b []byte) error {
 	return d.Tx([]byte{reg}, b)
 }
 
-// Shortcuts
+// ReadStruct writes the register number to the I²C bus, then reads data
+// marshalled.
+//
+// It uses reflection underneath.
+//
+// For slice of bytes, use ReadRegRaw(), for integers, use ReadRegUnitNN().
+func (d *DevReg) ReadRegStruct(reg byte, b interface{}) error {
```

Comment:
Why is this limited to a struct? binary.Read will happily read into other data structures... Or is it in order not to reimplement the binary.dataSize() function?